src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -3518,5 +3518,70 @@ public B() : this(1)
                     Diagnostic("ID0001", "B").WithLocation(7, 12)
                 });
         }
+
+        [Theory, CombinatorialData]
+        public async Task TestCategorizedAnalyzerDiagnosticsAsync(bool syntax, bool singleAnalyzer)
```

Comment:
@dotnet/roslyn-compiler The compiler changes are in the first 4 files in the PR. I have also separated them in a separate commit https://github.com/dotnet/roslyn/pull/45322/commits/7087a17656b00df61058f746b4ab050faeeedb3c for easier review. Thanks.

Reply:
FYI: I have actually moved all the compiler changes in this PR to a separate PR https://github.com/dotnet/roslyn/pull/45347. This PR should have no compiler changes once that PR is merged.